Do not use -Wno-format-contains-nul. clang doesn't understand it and prints
a warning:

warning: unknown warning option '-Wno-format-contains-nul' [-Wunknown-warning-option]

while gcc 4.2 prints an error:

cc1: error: unrecognized command line option "-Wno-format-contains-nul"

Use -Wno-format instead.

https://gcc.gnu.org/legacy-ml/gcc/2018-06/msg00216.html
--- configure.orig	2020-09-30 18:03:12.000000000 -0500
+++ configure	2020-10-01 11:53:08.000000000 -0500
@@ -18663,7 +18663,7 @@
 test "X${GCC}" = Xyes && \
     CFLAGS=`echo $CFLAGS -Wcast-align -Wmissing-prototypes \
 	-Wpointer-arith -Wshadow -Wstrict-prototypes -Wwrite-strings \
-	-Wno-format-contains-nul -fno-strict-aliasing -Wstrict-aliasing=2`
+	-Wno-format -fno-strict-aliasing -Wstrict-aliasing=2`
 
 
 
